core/src/test/java/io/kestra/core/repositories/AbstractExecutionRepositoryTest.java

```diff
@@ -498,6 +500,67 @@ protected void dailyStatistics() throws InterruptedException {
         assertThat(result.get(10).getExecutionCounts().get(State.Type.SUCCESS)).isEqualTo(1L);
     }
 
+    @Test
+    protected void dailyStatisticsEuropePragueMidnight() throws InterruptedException {
+        // Prepare two executions around midnight Europe/Prague: one before midnight and one after
+        ZoneId prague = ZoneId.of("Europe/Prague");
+
+        ZonedDateTime dayBeforeMidnight = ZonedDateTime.of(2025, 10, 25, 23, 30, 0, 0, prague);
+        ZonedDateTime dayAfterMidnight = ZonedDateTime.of(2025, 10, 26, 0, 30, 0, 0, prague);
+
+        // Build executions with explicit start dates without using Mockito.spy (avoid inline mock maker)
+        Execution before = Execution.builder()
+            .id(IdUtils.create())
+            .namespace(NAMESPACE)
+            .tenantId(MAIN_TENANT)
+            .flowId(FLOW)
+            .flowRevision(1)
+            .state(new State(State.Type.SUCCESS, List.of(new State.History(State.Type.SUCCESS, dayBeforeMidnight.toInstant()))))
+            .taskRunList(List.of())
+            .build();
+
+        Execution after = Execution.builder()
+            .id(IdUtils.create())
+            .namespace(NAMESPACE)
+            .tenantId(MAIN_TENANT)
+            .flowId(FLOW)
+            .flowRevision(1)
+            .state(new State(State.Type.SUCCESS, List.of(new State.History(State.Type.SUCCESS, dayAfterMidnight.toInstant()))))
+            .taskRunList(List.of())
+            .build();
+
+        executionRepository.save(before);
+        executionRepository.save(after);
+
+        ZonedDateTime start = dayBeforeMidnight.truncatedTo(ChronoUnit.DAYS);
+        ZonedDateTime end = dayAfterMidnight.plusDays(1).truncatedTo(ChronoUnit.DAYS);
+
+        // Wait up to 2 seconds for DB to reflect saved executions using Awaitility
+        List<DailyExecutionStatistics> result = Awaitility.await()
```

Comment:
Saving in DB is synchronous so it should not be needed.
Does it really pass without it? Something strange should happen in this case